src/handlers/v1/markets/[chainId]/[marketId]/zap-supply/index.ts

```diff
@@ -0,0 +1,207 @@
+    }
+
+    const service = new Service(chainId);
+
+    let marketInfo: MarketInfo;
+    try {
+      marketInfo = await service.getMarketInfo(marketId, account);
+    } catch (err) {
+      throw newInternalServerError(err);
+    }
+    const { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR } = marketInfo;
+    const currentPosition = { utilization, healthRate, liquidationThreshold, borrowUSD, collateralUSD, netAPR };
+
+    let targetTokenAmount = '0';
+    const logics: GetZapSupplyQuotationResponseBody['logics'] = [];
+    let approvals: GetZapSupplyQuotationResponseBody['approvals'] = [];
+    let permitData: GetZapSupplyQuotationResponseBody['permitData'];
+    let targetPosition = currentPosition;
+    if (event.body.sourceToken && event.body.targetToken && event.body.amount && Number(event.body.amount) > 0) {
+      const { amount, slippage } = event.body;
+      const {
+        baseToken,
+        baseTokenPrice,
+        supplyAPR,
+        supplyUSD,
+        borrowAPR,
+        borrowCapacityUSD,
+        liquidationLimit,
+        collaterals,
+      } = marketInfo;
+      const sourceToken = common.Token.from(event.body.sourceToken);
+      const targetToken = common.Token.from(event.body.targetToken);
+
+      // 1. check supply collateral or base token
+      const targetCollateral = collaterals.find(({ asset }) => asset.is(targetToken.unwrapped));
+      if (!targetCollateral && !targetToken.is(baseToken)) {
+        throw newHttpError(400, { code: '400.5', message: 'target token is not collateral nor base' });
+      }
+
+      // 2. new and append swap token logic
+      if (sourceToken.is(targetToken)) {
+        targetTokenAmount = amount;
+      } else {
+        const quotation = await apisdk.protocols.paraswapv5.getSwapTokenQuotation(chainId, {
+          input: { token: sourceToken, amount },
+          tokenOut: targetToken,
+          slippage,
+        });
+        targetTokenAmount = quotation.output.amount;
+        logics.push(apisdk.protocols.paraswapv5.newSwapTokenLogic(quotation));
+      }
+
+      // 3. new and append compound v3 supply logic
+      if (targetToken.is(baseToken)) {
+        if (!new BigNumberJS(borrowUSD).isZero()) {
+          throw newHttpError(400, { code: '400.6', message: 'borrow USD is not zero' });
+        }
+        const cToken = await service.getCToken(marketId);
+        const supplyBaseQuotation = await apisdk.protocols.compoundv3.getSupplyBaseQuotation(chainId, {
+          marketId,
+          input: {
+            token: targetToken,
+            amount: targetTokenAmount,
+          },
+          tokenOut: cToken,
+        });
+        logics.push(
+          apisdk.protocols.compoundv3.newSupplyBaseLogic({
+            input: supplyBaseQuotation.input,
+            output: supplyBaseQuotation.output,
+            marketId: supplyBaseQuotation.marketId,
+            balanceBps: common.BPS_BASE,
+          })
+        );
```

Comment:
Can we simplify it like this?
```suggestion
        logics.push(
          apisdk.protocols.compoundv3.newSupplyBaseLogic({
            ...supplyBaseQuotation,
            balanceBps: common.BPS_BASE,
          })
        );
```